allocation.py

```diff
@@ -22,6 +27,13 @@ def get_rec_name(self, name):
     def search_rec_name(cls, name, clause):
         return [('employee.rec_name',) + tuple(clause[1:])]
 
+    def get_company(self, name):
```

Comment:
I think it should be an on_change_with() that depends on "work" and "_parent_work.company"